client/src/legacy/LeftAndMain.Tree.js

```diff
@@ -162,6 +168,10 @@ $.entwine('ss.tree', function($){
       }
     },
 
+    canMove: function(data) {
```

Comment:
The default implementation should be a promise so that it can be `await`ed above:
```
    canMove: Promise.resolve(true);
```